### PR TITLE
[consensus] Proposal generator: fetch only to round-up over max_to_execute

### PR DESCRIPTION
**Pull request by igor-aptos**
## Description

When due to backpressure we want to create small blocks - i.e. 20 transactions, allow QS to only pull minimal number of batches to have at least 20 transactions, and use max_to_execute to cut afterwards

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by bchocho — APPROVED** [review]
LGTM. Do you think a counter or log that distinguishes whether the soft max or max (or both?) were hit would be useful?

File: config/src/config/consensus_config.rs [hunk 1]
```diff
@@ -228,6 +231,8 @@ impl Default for ConsensusConfig {
                 percentile: 0.5,
                 target_block_time_ms: 250,
                 min_block_time_ms_to_activate: 100,
+                // allow at least two spreading group from reordring in a single block, to utilize paralellism
```
Comment (vusirikala) [inline config/src/config/consensus_config.rs:234]: What is a spreading group?

Comment (vusirikala) [reply config/src/config/consensus_config.rs:234]: nit:reordering

**Review by vusirikala — APPROVED** [review]

File: config/src/config/consensus_config.rs (hunk 1 above)
Comment (igor-aptos) [reply config/src/config/consensus_config.rs:234]: https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-68.md

**Comment by igor-aptos** [comment]
I think overall block size, and overall limits should be enough

**Comment by igor-aptos** [comment]
forge test succeeded - https://github.com/aptos-labs/aptos-core/actions/runs/10276146477 , with metrics and logs looking better, and no latency increase. landing

**Pull request merged**